### ISSUE TITLE
checkout.session.completed 500 internal server error

### ISSUE THREAD

**Issue by allion-randika** [enupal/stripe#384]
When I am using the payment link to proceed with the payment, the payment is completed and can see in the stripe dashboard but not tracked in my CraftCMS backend.

### Additional info

- Craft version: 4
- Plugin version: 5.5

**Comment by andrelopez** [comment]
Hi @allion-randika Are you testing locally? could you please confirm if you're using the [Stripe CLI ](https://docs.enupal.com/stripe-payments/stripe-payment-forms/webhook.html#test-webhooks-locally)to redirect webhooks to local env?

**Comment by andrelopez** [comment]
Please feel free to re-open if needed

**Comment by romainpoirier** [comment]
I’m still affected on Craft CMS 4.17.x with enupal/stripe 5.5.2.

I traced the 500s on `checkout.session.completed` to this path:
- webhook reads `payment_intent.metadata['stripe_payments_form_id']`
- for some sessions (notably Stripe Payment Links / non-Enupal-origin flows), metadata is empty
- plugin then calls `PaymentForms::getPaymentFormById(null)` and throws TypeError, returning HTTP 500

As a temporary mitigation in production, I added a guard:
- if `stripe_payments_form_id` is missing, log warning, skip order creation for that event path, return HTTP 200.

After replaying failed deliveries, `delivery_success=false` is now 0 for the endpoint.
The replayed legacy events were acknowledged but not converted into Enupal orders (no matching Enupal metadata to map them).

Could this defensive handling be added upstream in the plugin?
